src/internal/tarutil/util.go

```diff
@@ -277,3 +277,27 @@ func Export(storageRoot string, w io.Writer, opts ...ExportOption) (retErr error
 		})
 	})
 }
+
+// Reader converts a set of files to a tar stream.
+// TODO: Probably should just go to disk for this.
+func Reader(files []File) (io.Reader, error) {
```

Comment:
nit: NewReader, or MakeReader for constructors.  Reserves Reader for an exported type later.